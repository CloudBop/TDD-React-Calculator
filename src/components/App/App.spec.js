import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import Calculator from '../Calculator/Calculator';

describe('App', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<App />));

  // add a snapshot tests as the first test in each spec file. This creates a pattern of placing the generic tests (snapshot, shallow render) that appear in each spec file above the component specific tests
  it('Should compare against snapshot <App/>', () => expect(wrapper).toMatchSnapshot())
  // similar to above but no shapshot
  // it('should render a <div />', () => {
  //   expect(wrapper.find('div').length).toEqual(1);
  // });
  
  it('should render the Calculator Component', () => {
    expect(wrapper.containsMatchingElement(<Calculator />)).toEqual(true);
  });
  
});