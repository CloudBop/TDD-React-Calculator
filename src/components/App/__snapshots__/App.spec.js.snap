// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should compare against snapshot 1`] = `ShallowWrapper {}`;
